src/ir/cost.h

```diff
@@ -684,6 +684,10 @@ struct CostAnalyzer : public OverriddenVisitor<CostAnalyzer, CostType> {
   CostType visitStringConcat(StringConcat* curr) {
     return 10 + visit(curr->left) + visit(curr->right);
   }
+  CostType visitStringEq(StringEq* curr) {
+    // This assumes strings are interned in the engine.
```

Comment:
I would be surprised if all strings were interned! This seems like it should be modeled as being very expensive since it's cost is unbounded.

Reply:
I think the VM will intern them if they are used enough, at least in some cases, but I don't really know. JS string implementations are quite flexible. I guess we shouldn't strongly assume anything here, but like with concat, I don't think this can be a problem in general - there is no equivalent code that would be compared to this.

Reply:
I adjusted it to 3 (in between 1 and 6 for interned and not), and added a comment.